src/Orc.DataAccess/Database/Extensions/DbProviderInfoExtensions.cs

```diff
@@ -1,19 +1,19 @@
 ﻿namespace Orc.DataAccess.Database
 {
-    using Catel;
+    using System;
 
     public static class DbProviderInfoExtensions
     {
-        public static DbConnectionString CreateConnectionString(this DbProviderInfo dbProviderInfo, string? connectionString = null)
+        public static DbConnectionString? CreateConnectionString(this DbProviderInfo dbProviderInfo, string? connectionString = null)
         {
-            Argument.IsNotNull(() => dbProviderInfo);
+            ArgumentNullException.ThrowIfNull(dbProviderInfo);
 
-            return dbProviderInfo.GetProvider()?.CreateConnectionString(connectionString);
+            return dbProviderInfo.GetProvider()?.CreateConnectionString(connectionString ?? string.Empty);
```

Comment:
since we are defaulting to string.Empty in case of null, why not change connectionString parameter to "" as default value and non-nullable?